internal/provider/resource_vault_cluster.go

```diff
@@ -169,8 +175,20 @@ func resourceVaultClusterCreate(ctx context.Context, d *schema.ResourceData, met
 
 	publicEndpoint := d.Get("public_endpoint").(bool)
 
-	// TODO: Tier is hard-coded for now, but eventually will be required input on the resource.
-	tier := vaultmodels.HashicorpCloudVault20201125TierDEV
+	//tiers from here: https://github.com/hashicorp/hcp-sdk-go/blob/f7f0dd4b49fd46758c2d2c9a151fe3a70e5e271e/clients/cloud-vault-service/preview/2020-11-25/models/hashicorp_cloud_vault20201125_tier.go#L21
+	t, err := d.Get("tier").(string)
+	switch t {
+	 case "development":
+			 tier := vaultmodels.HashicorpCloudVault20201125TierDEV
+	 case "small":
```

Comment:
I see that the Consul implementation separates tier and size which seems more future proof if there would be another tier introduced. Also it would be a more consistent codebase between the two services. Should we proceed with this solution and do a patch later when the vault models are updated?

Reply:
I added it as a TODO